### PR TITLE
Activity -  Refactor code using SOLID principle

### PR DESCRIPTION
**Pull request by Kingjosh007**
In this activity, @gracekabaghe and I had to refactor the code to meet [these requirements](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_refactor.md). 
This is an overview of what we did:
- We created a file called `lists.rb` where we used the OOP concepts like inheritance to refactor the way we display lists.
- We created a file called `create.rb` which contains refactor code for the create_methods
- We created custom `-Listing` and `-Creator` classes for each type of object (Person, Book, Rental)
- The previous actions allowed us to reduce the number of lines of code in `main.rb` file.

### CODE REVIEW

File: create.rb [hunk 1]
```diff
@@ -0,0 +1,133 @@
+require_relative 'book'
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'rental'
+require_relative 'classroom'
+require_relative 'colors_utils'
+
+# rubocop: disable Style
```
Comment (hiromataba) [inline create.rb]: - [x] Linting is important to reduce errors and improve the overall quality of your code  so I think fixing them would be much better than disabling them

File: lists.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
+require_relative 'book'
+require_relative 'person'
+require_relative 'rental'
+require_relative 'colors_utils'
+
+# rubocop: disable Layout/LineLength
+# rubocop: disable Style
```
Comment (hiromataba) [inline lists.rb]: - [x] Linting is important to reduce errors and improve the overall quality of your code  so I think fixing them would be much better than disabling them

**Review by hiromataba — CHANGES_REQUESTED** [review]
Hi :wave: :wave: @Kingjosh007 and @gracekabaghe , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
### To highlight :1st_place_medal: 
- Correct git-flow :heavy_check_mark: 
- Nice PR description and title :heavy_check_mark: 
- Your app is running correctly and you have a very beautiful UI :heavy_check_mark: 
- Your README file looks very professional :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: create.rb (hunk 1 above)
Comment (Kingjosh007) [reply create.rb]: Resolved.

File: lists.rb (hunk 2 above)
Comment (Kingjosh007) [reply lists.rb]: Resolved.

File: create.rb (hunk 1 above)
Comment (rloterh) [reply create.rb]: This issue is well resolved! ✔️

File: lists.rb (hunk 2 above)
Comment (rloterh) [reply lists.rb]: This issue is well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Grace & KJC, 

Your project is complete! Great job for making the necessary changes recommended in the previous review. There is nothing else to say other than... it's time to merge it 🍾 :shipit:. Congratulations! 🎉🥂

<br>
Cheers and Happy coding!👏 💻 🍷 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 1 more limited review per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**